charts/tidb-operator/templates/scheduler-rbac.yaml

```diff
@@ -23,50 +23,23 @@ metadata:
     helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+"  "_" }}
 rules:
 - apiGroups: [""]
-  resources: ["pods", "services", "configmaps", "replicationcontrollers", "persistentvolumeclaims", "endpoints"]
-  verbs: ["get", "list", "watch"]
-- apiGroups: [""]
-  resources: ["pods/binding"]
-  verbs: ["create"]
-- apiGroups: [""]
-  resources: ["pods/status"]
-  verbs: ["patch", "update"]
-- apiGroups: [""]
-  resources: ["endpoints", "events"]
-  verbs: ["get", "list", "watch", "create", "update", "patch"]
-- apiGroups: ["apps"]
-  resources: ["statefulsets"]
-  verbs: ["get", "list", "watch"]
-- apiGroups: ["policy"]
-  resources: ["poddisruptionbudgets"]
-  verbs: ["get", "list", "watch"]
-- apiGroups: ["apps", "extensions"]
-  resources: ["replicasets"]
-  verbs: ["get", "list", "watch"]
-- apiGroups: ["storage.k8s.io"]
-  resources: ["storageclasses"]
+  resources: ["pods", "configmaps"]
   verbs: ["get", "list", "watch"]
 - apiGroups: [""]
   resources: ["nodes"]
-  verbs: ["get", "list", "watch"]
-- apiGroups: [""]
-  resources: ["persistentvolumes"]
-  verbs: ["get", "list", "watch", "update"]
+  verbs: ["get", "list"]
 - apiGroups: ["pingcap.com"]
   resources: ["tidbclusters"]
   verbs: ["get"]
 - apiGroups: [""]
   resources: ["persistentvolumeclaims"]
   verbs: ["get", "list", "update"]
-# followng permissions are required if CSINodeInfo/AttachVolumeLimit features are enabled
-- apiGroups:
-  - storage.k8s.io
-  resources:
-  - csinodes
-  verbs:
-  - get
-  - list
-  - watch
+- apiGroups: ["storage.k8s.io"]
+  resources: ["storageclasses"]
+  verbs: ["get", "list", "watch"]
+- apiGroups: [""]
+  resources: ["endpoints"]
+  verbs: ["get", "list", "watch", "update"]
```

Comment:
I guess tidb-scheduler/tidb-scheduler does not need permissions of storageclasses and endpoints.

Reply:
Yep, these permissions are used for kube-scheduler. In my test case,  kube-scheduler log got this error:
`E1223 02:11:01.851511       1 leaderelection.go:306] error retrieving resource lock tidb-operator/tidb-scheduler: endpoints "tidb-scheduler" is forbidden: User "system:serviceaccount:tidb-operator:tidb-scheduler" cannot get resource "endpoints" in API group "" in the namespace "tidb-operator"`
and I check endpoint permission not in system:kube-scheduler (cluster-role), so I add it to tidb-scheduler role/clusterrole.

Reply:
And storageclass permissions already in the clusterrole system:volume-scheduler, so I will remove it from tidb-scheduler role.

Reply:
can you paste the full output of `system:kube-scheduler` in your cluster?

I guess it is because we uses `tidb-scheduelr` (and configurable) other than kube-scheduler as leader election endpoint name, so we need permissions on `tidb-scheduler` permissin.

Because it's configurable, we need extra permissions (`delete/get/patch/update`) for '*' endpoints.

By the way, please add some comments for these special cases:

- ConfigMap permission for `--policy-configmap`
- Extra permissions for endpoints other than `kube-scheduler`

Reply:
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  annotations:
    rbac.authorization.kubernetes.io/autoupdate: "true"
  labels:
    kubernetes.io/bootstrapping: rbac-defaults
  name: system:kube-scheduler
rules:
- apiGroups:
  - ""
  resources:
  - events
  verbs:
  - create
  - patch
  - update
- apiGroups:
  - ""
  resources:
  - endpoints
  verbs:
  - create
- apiGroups:
  - ""
  resourceNames:
  - kube-scheduler
  resources:
  - endpoints
  verbs:
  - delete
  - get
  - patch
  - update
- apiGroups:
  - ""
  resources:
  - nodes
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - ""
  resources:
  - pods
  verbs:
  - delete
  - get
  - list
  - watch
- apiGroups:
  - ""
  resources:
  - bindings
  - pods/binding
  verbs:
  - create
- apiGroups:
  - ""
  resources:
  - pods/status
  verbs:
  - patch
  - update
- apiGroups:
  - ""
  resources:
  - replicationcontrollers
  - services
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - apps
  - extensions
  resources:
  - replicasets
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - apps
  resources:
  - statefulsets
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - policy
  resources:
  - poddisruptionbudgets
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - ""
  resources:
  - persistentvolumeclaims
  - persistentvolumes
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - authentication.k8s.io
  resources:
  - tokenreviews
  verbs:
  - create
- apiGroups:
  - authorization.k8s.io
  resources:
  - subjectaccessreviews
  verbs:
  - create

Reply:
I see, please update permission of endpoints from `get/list/watch/update` to `delete/get/patch/update`.